src/ui/shared/WindowContext.ts

```diff
@@ -0,0 +1,50 @@
+export type WindowType = 'popup' | 'tab' | 'dialog';
+export type WindowLayout = 'column' | 'page';
+
+export enum WindowParam {
+  windowType = 'windowType',
+  windowLayout = 'windowLayout',
+  mode = 'mode',
+}
+
+function getParam<T>(param: WindowParam) {
+  const url = new URL(window.location.href);
+  return url.searchParams.get(param) as T | null;
+}
+
+export class WindowContext {
+  readonly windowType: WindowType;
+  readonly windowLayout: WindowLayout;
+
+  private constructor(windowType: WindowType, windowLayout: WindowLayout) {
+    this.windowType = windowType;
+    this.windowLayout = windowLayout;
+  }
+
+  static get() {
```

Comment:
Why not create a singleton entity here and make constructor public?

Reply:
Sometimes we want to be able to just import the types from this file, without accessing `window` (for example, in background script). To me it is just safer to call the `get` function explicitly, when needed. This way we can be sure we're not accessing anything we didn't want to access. We could move the types somewhere, but they seem to belong here, so...

Reply:
Also this way we can be sure it is always up to date (since we read query string parameters every time we're calling the `get` fn)

Reply:
If I understand right, importing types should be safe with import type {} syntax

Reply:
Good moment about the up-to-date state.
Let me think

Reply:
I think, we can still use just a Constuctor everywhere.
So const windowContext = new WindowContext() should work fine everywhere in ui

Reply:
> importing types should be safe with import type {} syntax
Right, it's fine until we need to import something that's not a type, like enum WindowParam -- that's what led me to the current design. Moving WindowParam to another place just didn't feel right

> So const windowContext = new WindowContext() should work fine everywhere in UI
Yes, I think so. At least for now, it should work

Reply:
✅

Reply:
Not relevant anymore